### DISCUSSION TITLE
Field OPP-070-notice

### DISCUSSION THREAD

**Discussion by rengit62** [OP-TED/eForms-SDK#121] [Q&A] [answered]
OPP-070-notice has got xpathRelative" : "efac:NoticeSubType/cbc:SubTypeCode without any @listName
but In the sample with subtype 4 (pin-only_24,xml for example ) has got the attribute :  listName="notice-subtype".
Question : as the field has got "type" : "code", "codeList" : {
      "value" : {
        "id" : "notice-subtype",
        "type" : "flat"
      },
is it 'implicit' that the field has got an attribute called "listName" ? This is valid for all fields ?
Thanks

**Comment by YvesJo** [comment, marked as answer]
Hi,
Yes, for any XML element of type Code the listName attribute is 'implicitely' required.

**Reply by rengit62** [reply]
Thanks Yves but after I wrote I saw these fields on Metadata Section (pin-only_24.-maximal.xml)
 
	<cbc:NoticeTypeCode listName="planning">pin-only</cbc:NoticeTypeCode> 
BT-02-notice should correspond to cbc:NoticeTypeCode with the value "pin-only"  while 
BT-03-notice with xpathRelative" : "cbc:NoticeTypeCode/@listName should be the attribute called listName with value "planning" of the tag cbc:NoticeTypeCode. Both BT-02-notice and BT-03-notice have got "type" : "code"
I am a little "confused" . I do not know if am doing well , I am writing a java main that should be 'agnostic' of the actual types of notices.
So I have to know the rules ..
Bye
Renato

**Reply by YvesJo** [reply]
Hi Renato,
Sorry, I didn't see you remaining question. 
If you would like to know what BT-02 and BT-03 should be, you may have a look at the following figure: 
![image](https://user-images.githubusercontent.com/42802091/198553585-537c180e-df15-41f8-ab52-5847b32088c5.png)
![image](https://user-images.githubusercontent.com/42802091/198553803-5076ca6b-73ed-422d-bd3b-8ad44a3869f8.png)
You may also reverse engineer the codelists, there are notice type subsets for each form type (e.g. [competition](https://github.com/OP-TED/eForms-SDK/blob/develop/codelists/notice-type_competition.gc) ) and notice subtype subsets for each notice type (e.g.  [can-desg](https://github.com/OP-TED/eForms-SDK/blob/develop/codelists/notice-subtype_can-desg.gc) )
Also consider that for any notice subtype change is also a valid form type (of course if the notice is about a change).
I wish this will answer your question.
